### ISSUE TITLE
Simplify Recipe Model

### ISSUE THREAD

**Issue by ethanmick** [plate0/platezero#363]
Can we simplify the recipe model and the versioning? Should we? Looking at the data model now, we have:

```
ingredient_lines
ingredient_list_lines
ingredient_lists
preheats
procedure_lines
procedure_list_lines
procedure_lists
recipe_branches
recipe_collaborators
recipe_durations
recipe_version_ingredient_lists
recipe_version_preheats
recipe_version_procedure_lists
recipe_versions
recipe_yields
recipes
```

That seems like a lot! I know we support versioning, and a lot of `git` like functionality, but after we've had this for a while, do we _need_ it all? A recipe, at its simplest, can be modeled as:

```
recipes

## Meta Data
recipe_yields
preheats
recipe_durations

## Instructions
ingredient_lines
procedure_lines
```

We do support multiple ingredient and procedure _sections_, the `*_lists`. That's probably good to have, a lot of recipes make use of that.

Do we need versioning? Is our implementation the best of that? My hunch says it might be worth simplifying the entire thing down to a `previous` key, or one table that tracks all the versions of a recipe, with the information being more duplicated. Thoughts?

**Comment by benburwell** [comment]
Agreed. I don't think we need the complex model we started with. We might even simplify it down to the procedure lines just being a giant markdown field.

Something like

```sql
create table recipes (
  id serial primary key,
  name varchar
)

create table recipe_versions (
  recipe_id int foreign key references recipes (id),
  yield varchar,
  -- preheats, durations, ingredients i guess we'd still need a separate table for since its 1:n
  procedure varchar
)
```

With the basic idea being that everything is stored as a `recipe_version`, and we just look at the most recently created timestamp for the "current" version?
